modules/core/src/config.ts

```diff
@@ -31,46 +31,51 @@ export interface Tokens {
 }
 
 // Get the list of ERC-20 tokens from statics and format it properly
-const formattedErc20Tokens = coins.filter((coin: BaseCoin) => {
-  return coin instanceof Erc20Coin;
-}).map((token: Erc20Coin): Erc20TokenConfig => {
-  return {
-    type: token.name,
-    coin: token.network.type === NetworkType.MAINNET ? 'eth' : 'teth',
-    network: token.network.type === NetworkType.MAINNET ? 'Mainnet' : 'Testnet',
-    name: token.fullName,
-    tokenContractAddress: token.contractAddress.toString().toLowerCase(),
-    decimalPlaces: token.decimalPlaces,
-  };
-});
+const formattedErc20Tokens = coins.reduce((acc, coin) => {
```

Comment:
the `filter`/`map` variant looked much nicer

what do you think about
```
(coins.filter(...) as Erc20Coin[])
.map(...)
```
?

Reply:
Well, I suppose that's subjective since I find the reduce version a bit cleaner, but I agree that there is room for improvement here. Let me take another pass at it.